app/src/main/java/com/duckduckgo/app/browser/localstorage/LocalStorageManager.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.localstorage
+
+import android.content.Context
+import com.duckduckgo.app.pixels.remoteconfig.AndroidBrowserConfigFeature
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import com.squareup.anvil.annotations.ContributesTo
+import dagger.Module
+import dagger.Provides
+import dagger.SingleInstanceIn
+import java.io.File
+import java.nio.charset.StandardCharsets
+import javax.inject.Inject
+import javax.inject.Provider
+import kotlinx.coroutines.runBlocking
+import org.iq80.leveldb.DB
+import org.iq80.leveldb.Options
+import org.iq80.leveldb.impl.Iq80DBFactory.factory
+import timber.log.Timber
+
+interface LocalStorageManager {
+    fun clearLocalStorage()
+}
+
+@ContributesBinding(AppScope::class)
+class DuckDuckGoLocalStorageManager @Inject constructor(
+    private val databaseProvider: Provider<DB>,
+    private val androidBrowserConfigFeature: AndroidBrowserConfigFeature,
+    private val localStorageSettingsJsonParser: LocalStorageSettingsJsonParser,
+) : LocalStorageManager {
+
+    private var domains = emptyList<String>()
+    private var matchingRegex = emptyList<String>()
+
+    override fun clearLocalStorage() = runBlocking {
+        val settings = androidBrowserConfigFeature.localStorage().getSettings()
+        val localStorageSettings = localStorageSettingsJsonParser.parseJson(settings)
+
+        domains = localStorageSettings.domains.list
+        matchingRegex = localStorageSettings.matchingRegex.list
+
+        Timber.d("LocalStorageManager: Allowed domains: $domains")
+        Timber.d("LocalStorageManager: Matching regex: $matchingRegex")
```

Comment:
Mental note, we need to migrate all logs to `logcat`. String interpolation will be executed in production even if logs are disabled. Which is wasteful and in hot paths perf can even be noticeable if the strings are big

No need to do anything here